### PR TITLE
feat(controller): online eval support gc

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

Online eval support garbage collection
- removes the workloads that reached the max TTL
- removes the oldest workload when there are pending workloads
  - prevent removing the workloads that not reached the min TTL

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java [hunk 1]
```diff
@@ -126,16 +140,23 @@ public ModelServingVo create(
                 .modelVersionId(model.getId())
                 .jobStatus(JobStatus.CREATED)
                 .resourcePool(resourcePool)
+                .lastVisitTime(new Date())
                 .build();
 
-        modelServingMapper.add(entity);
+        long id;
+        synchronized (this) {
+            modelServingMapper.add(entity);
```
Comment (goldenxinxing) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java:148]: Is it more clearly with the name of 'modelServingMapper.insertIgnore(...)'?

File: server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java [hunk 2]
```diff
@@ -126,16 +140,23 @@ public ModelServingVo create(
                 .modelVersionId(model.getId())
                 .jobStatus(JobStatus.CREATED)
                 .resourcePool(resourcePool)
+                .lastVisitTime(new Date())
                 .build();
 
-        modelServingMapper.add(entity);
+        long id;
+        synchronized (this) {
```
Comment (goldenxinxing) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java:147]: Is the synchronized block written to ensure idempotence? if insert is ignore for exception, it seems unnecessary to use sync block

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java:147]: This block is for preventing the race condition with the GC thread, but the synchronized range in the GC thread is wrong, I will change it

File: server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java (hunk 1 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java]: Done in 8f65c60

File: server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java (hunk 2 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/job/ModelServingService.java:145]: Done in 8f65c60

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**